Examples/package-info/Sources/package-info/main.swift

```diff
@@ -1,10 +1,23 @@
 import Workspace
 
-let swiftpm = SwiftPMHelper(rootPackage: localFileSystem.currentWorkingDirectory!)
-let workspace = swiftpm.createWorkspace()
+/// The path of the package.
+let package = localFileSystem.currentWorkingDirectory!
 
+/// Determine the directory that contains SwiftPM's PackageDescription libs.
+let libDir = AbsolutePath(#file).appending(RelativePath("../../../.build/.bootstrap/lib/swift/pm"))
```

Comment:
@SDGGiesbrecht I figured UserToolchain/Destination was wrong approach for what you're trying to achieve. Those objects are meant for performing builds and manifest/package graph loading just requires location of the SwiftPM libs. It is also wrong to look for libs in an installed toolchain since SwiftPM's libs and the version of SwiftPM you're using may not agree. We shouldn't encourage that either so I think it's much safer to expect clients to pass in the lib dir.

Reply:
I think this is almost what I originally had. You can see my current code [here](https://github.com/SDGGiesbrecht/SDGSwift/blob/9b22a986279e38436ee7b2889cc8e629488be7ac/Sources/SDGSwiftPackageManager/PackageRepository.swift#L79-L152) if it is helpful to you. The history of it is kinda foggy in my memory, but I think I pulled the extra toolchain stuff into the equation precisely because it could accurately provide the location of the libs based on the location of `swift`. `swift` is easy to find in a resilient way as a client, but the relative location of `pm` is an implementation detail I no longer wanted to rely on. For what it’s worth, my code that searches for `swift` will only accept the version(s) matching the SwiftPM it was built with, so mismatched libraries are not an issue. (Though it would be nice if someday SwiftPM handled that on its own.) Ultimately I think in general that saying mismatched system installs aren’t supported is better than asking clients to poke around in the implementation details of the Swift toolchain layout.

That means I would prefer this line read something like...

```swift
let swift = AbsolutePath(#file)
    .appending(RelativePath("../../../.build/.bootstrap/swift"))
// A client might point at “/usr/bin/swift”.
```
...and the rest followed from there.

***

> It is also wrong to look for libs in an installed toolchain since SwiftPM's libs and the version of SwiftPM you're using may not agree.

I hear what you are saying, but may I ask where else you expect a package client to get them from?

Reply:
In theory, we should bundle the libs as part of libSwiftPM but SwiftPM is not powerful enough to build them today. One we can do that though, the clients will no longer need to provide any kind of path for manifest/package graph operations.

> `swift` is easy to find in a resilient way as a client

Do you mean the absolute path and not the OS/swiftenv shim? If yes, then it seems reasonable to accept the toolchain path as input and SwiftPM can be responsible for figuring out the lib path from there.

Reply:
> Do you mean the absolute path and not the OS/swiftenv shim?

Yes, that is what I meant, the client having already resolved its location by querying whatever other tools about where they may have hidden the real toolchain away (e.g. by calling `xcrun --find swift`).

Reply:
👍